File: academic-system/src/main/java/org/example/repository/RepositorioTurmaXml.java
package org.example.repository;

import org.example.model.Turma;
import org.example.model.Avaliacao;
import java.util.List;
import java.io.FileWriter;
import java.io.PrintWriter;
import java.io.IOException;
import org.example.exception.ExcecaoSistemaAcademico;

public class RepositorioTurmaXml implements RepositorioTurma {
    
    private String arquivo = "turmas.xml";

    public RepositorioTurmaXml() {}
    public RepositorioTurmaXml(String arquivo) { this.arquivo = arquivo; }

    @Override
    public void salvarTodas(List<Turma> turmas) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(arquivo))) {
            writer.println("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
            writer.println("<sistemaAcademico>");
            writer.println("  <turmas>");
            for (Turma turma : turmas) {
                writer.println("    <turma>");
                writer.println("      <codigo>" + turma.getCodigo() + "</codigo>");
                writer.println("      <titulo>" + turma.getTitulo() + "</titulo>");
                writer.println("      <avaliacoes>");
                for (Avaliacao avaliacao : turma.getAvaliacoes()) {
                    writer.println("        <avaliacao>");
                    writer.println("          <tipo>" + avaliacao.getClass().getSimpleName() + "</tipo>");
                    writer.println("          <valor>" + avaliacao.getValor() + "</valor>");
                    writer.println("          <peso>" + avaliacao.getPeso() + "</peso>");
                    writer.println("        </avaliacao>");
                }
                writer.println("      </avaliacoes>");
                writer.println("    </turma>");
            }
            writer.println("  </turmas>");
            writer.println("</sistemaAcademico>");
        } catch (IOException e) {
            throw new ExcecaoSistemaAcademico("Erro ao salvar dados no arquivo XML: " + e.getMessage());
        }
    }
}

File: academic-system-csharp/src/SistemaAcademico/Repositories/RepositorioTurmaXml.cs
using SistemaAcademico.Models;
using SistemaAcademico.Exceptions;

namespace SistemaAcademico.Repositories;

// Equivalente à classe Java: RepositorioTurmaXml implements RepositorioTurma
// A geração de XML é manual (string building), idêntica à implementação Java original.
// Não usamos XmlSerializer para manter a fidelidade ao formato de saída original.
public class RepositorioTurmaXml : IRepositorioTurma
{
    private string arquivo = "turmas.xml";

    public RepositorioTurmaXml() { }

    public RepositorioTurmaXml(string arquivo)
    {
        this.arquivo = arquivo;
    }

    public void SalvarTodas(List<Turma> turmas)
    {
        try
        {
            using StreamWriter writer = new StreamWriter(arquivo);
            writer.WriteLine("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
            writer.WriteLine("<sistemaAcademico>");
            writer.WriteLine("  <turmas>");
            foreach (Turma turma in turmas)
            {
                writer.WriteLine("    <turma>");
                writer.WriteLine("      <codigo>" + turma.Codigo + "</codigo>");
                writer.WriteLine("      <titulo>" + turma.Titulo + "</titulo>");
                writer.WriteLine("      <avaliacoes>");
                foreach (Avaliacao avaliacao in turma.GetAvaliacoes())
                {
                    writer.WriteLine("        <avaliacao>");
                    writer.WriteLine("          <tipo>" + avaliacao.GetType().Name + "</tipo>");
                    writer.WriteLine("          <valor>" + avaliacao.Valor + "</valor>");
                    writer.WriteLine("          <peso>" + avaliacao.Peso + "</peso>");
                    writer.WriteLine("        </avaliacao>");
                }
                writer.WriteLine("      </avaliacoes>");
                writer.WriteLine("    </turma>");
            }
            writer.WriteLine("  </turmas>");
            writer.WriteLine("</sistemaAcademico>");
        }
        catch (IOException e)
        {
            throw new ExcecaoSistemaAcademico("Erro ao salvar dados no arquivo XML: " + e.Message);
        }
    }
}
